src/xdocs/writingchecks.xml

```diff
@@ -679,14 +688,25 @@ public class MethodLimitCheck extends AbstractCheck
           <a href="https://github.com/checkstyle/checkstyle/issues/10629">
           Issue #10629.</a>
         </li>
-        <li>To get valid violations, code has to be compilable.
-          If it is not, you can get hard to understand parse errors.</li>
         <li>You cannot determine the type of an expression.
         Example: "getValue() + getValue2()"</li>
         <li>You cannot determine the full inheritance hierarchy of type.</li>
-        <li>You cannot see the contents of other files. You have access to the
+      </ul>
+      <p>
+        In addition to Java files, there are similar limitations that apply to all
+        type of text files:
+      </p>
+      <ul>
+        <li>
+          Checkstyle cannot properly handle a file's binary
```

Comment:
I am not sure why, but I do not like `properly` here. Maybe `Checkstyle will not be able to parse...`?

Reply:
`not be able to parse` seems a bit excessive here. In terms of Java not being able to parse, it basically means the file will not be usable in checkstyle at all, like we got an exception on trying to work with it. BOM is not to that extreme. At the most, we will get extra garbage in the stream that should be ignored as it is just the header of the file. This was why I went with "properly handle". It can still be handled, but extra precautions will be needed.

Reply:
> Would you expect Checkstyle to consume/ manipulate/ ignore file contents? What if we wanted to have a check that creates a violation for using BOMs?

Basically Yes.

Checkstyle goes against text files, as our license header says, and even the name of the file we get the contents from is called `FileText`. The BOM is not part of the file's text. It is purely a header to describe the format of the rest of the file. If we wanted binary data, they should really not be using `FileText`. If we were going to support raw binary data and examining it, I would see us needing some intermediary before `FileText` like `RawFileData` or such.

Reply:
> t is purely a header to describe the format of the rest of the file. If we wanted binary data, they should really not be using FileText

This is also a good point.